### ISSUE TITLE
Switch all documentation over from GLPK to HiGHs

### ISSUE THREAD

**Issue by oxinabox** [COBREXA/COBREXA.jl#57]
## Target functionality

The docs on occasion reference using GLPK.
Use of GLPK should generally be discouraged at this point.

To quote the JuMP issue (they were talking about MILP but point stands for LP as well) https://github.com/jump-dev/JuMP.jl/issues/2878

> ... examples in the JuMP documentation use GLPK... However, it has the down-side of being the first solver that new users see when they read the docs.
> 
> We should switch ... examples to HiGHS because it's **faster, more robust, and being actively developed.**

(Emphasis mine)

To make kinda clear why consider FBA on yeast.
(I can make this a MWE if wanted but I don't think needed, its on [this SBML](https://github.com/SysBioChalmers/yeast-GEM/blob/main/model/yeast-GEM.xml))
Second runtime on each, so it is all compiled

```julia-repl
julia> @time sol_glpk = flux_balance_analysis(model, optimizer = GLPK.Optimizer);
  1.387430 seconds (838.36 k allocations: 49.660 MiB)
julia> @time sol_clp = flux_balance_analysis(model, optimizer = Clp.Optimizer);
  0.405093 seconds (805.17 k allocations: 49.605 MiB)
julia> @time sol_highs = flux_balance_analysis(model, optimizer = HiGHS.Optimizer);
  0.154727 seconds (827.32 k allocations: 50.077 MiB)
```

## Desired output

The docs should reference using HiGHs instead.

## Optional: Suggestions for implementation

Basically just change every reference to GLPK [found here](https://github.com/search?q=repo%3ACOBREXA%2FCOBREXA.jl%20GLPK&type=code) to be HiGHs.

**Comment by stelmo** [comment]
Hi there! I had no idea HiGHs was so much better! Thanks for bringing this to our attention. We will change our docs to reflect this :)

**Comment by exaexa** [comment]
Hi! Certainly thanks for bringing this up, but why exactly HiGHS? GLPK is used here on purpose not because it's fastest or most featureful, but because it's the least surprising one (if it fails, it fails predictably, and every other metabolic modelling package uses it as default as well). Requirement for picking the right solver for "real stuff" is known pretty well in the community, and in the end everyone is going to go with their fav Gurobi anyway :D

If there is no other benefit than speed we might want to go for something simpler instead (we already used Tulip at some point, right?). Other choices (possibly valid also for JuMP) include SCIP and ECOS. Anyway this is nothing against HiGHS, more like just checking if the choice isn't wider.

**Comment by exaexa** [comment]
aaaanyway, fished this out of history:

[fba-all.pdf](https://github.com/user-attachments/files/16390908/fba-all.pdf)

yeah let's do highs I guess

**Comment by oxinabox** [comment]
I agree everyone in academia will use Gurobi,
but if you can't get an academic license Gurobi is prohibitively expensive.
And if you are just getting into the area you might not a) know not all solvers are equal, b) might not have a Gurobi liscense.

Its not just speed HiGHs is:

> faster, more robust, and being actively developed.

A lot of the older open source solvers are largely unmaintained: its a problem (its a problem for all open source but especially for solvers). Fortunately they mostly just keep working.
@odow is the person to talk to about what's going on in that world.
In general, my feeling is HiGHs is now just the generally most sensibly solver to use for linear problems.

**Comment by odow** [comment]
There was essentially never a good reason to use GLPK. The only historical reason we used it in the JuMP docs was because it supported callbacks, and we wanted to minimize the number of unique solvers in the documentation (because installing and (pre)compilation was getting pretty expensive).

If you want an open-source (MI)LP solver, use HiGHS. Don't look anywhere else, and don't even really bother with benchmarks. The "being actively developed" outweighs everything else.

@jajhall: here's another use-case for you: "exa-scale metabolic modeling"

**Comment by jajhall** [comment]
Thanks @odow for enlightening another community. Also, COBREXA people, Oscar is right in referring to it as HiGHS, not HiGHs.

**Comment by exaexa** [comment]
I tried that in #60, looks like there's some small issues to be clarified, but could work.

> Its not just speed HiGHs is:
> 
> > faster, more robust, and being actively developed.
> 

That's a pretty strong claim, citation needed. (Does HiGHS have any other research/proofs out than the one referred from the website?)

I've got the same feeling that we should just go for HiGHS if it's feasible here (from #60 it might very well be), but I'd much rather have a proof in hand.

**Comment by odow** [comment]
> That's a pretty strong claim, citation needed

https://plato.asu.edu/ftp/lpopt.html
https://plato.asu.edu/ftp/milp.html
https://github.com/ERGO-Code/HiGHS/graphs/contributors

**Comment by exaexa** [comment]
Yeah. The flipped argument works for me too, except for the "minimal surprise" rule (which gets interesting in this community :D )

Let me work out what's wrong in #60 tests, other than that all good

**Comment by exaexa** [comment]
Hi all, the switch has just been finished, it should get deployed to the stable docs in next release. Thanks all for the info!

One more thing (just curious™): @oxinabox, how come you (as JuMP devs I guess) noticed this? You are using cobrexa for something, or just went through all JuMP packages and scanned for GLPK use?

**Comment by odow** [comment]
> One more thing (just curious™): @oxinabox, how come you (as JuMP devs I guess) noticed this?

@oxinabox is not a JuMP developer, but I think in this case, is an actual user of COBREXA :smile:

 * https://cellbauhaus.com
 * https://www.linkedin.com/posts/frames-catherine-white-46b9a035_im-happy-to-share-that-im-starting-a-new-activity-7222536626709700608-dNH0?utm_source=share&utm_medium=member_desktop

**Comment by oxinabox** [comment]
I am not a JuMP developer, but I am a long time user of JuMP.
And a very new (and confused) user of COBREXA.
my background isn't in systems bio (thus very confused, all the time).

**Comment by exaexa** [comment]
@oxinabox ah ok good to know. Pls feel free to ask on julia slack, we kinda concentrate in #sbml nowadays
